clayui.com/plugins/gatsby-plugin-svg-compile-sprite/gatsby-node.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * SPDX-FileCopyrightText: © 2018 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// Compiles SVG icons in directory `clay-css/src/images/icons` into a spritemap
+
+const fs = require('fs');
+const path = require('path');
+
+const generateSvgSprite = (pluginOptions) => {
+	const REGEX_FILE_EXT_SVG = /(flags-|.svg$)/g;
+	const REGEX_HTML_COMMENTS = /<!--(.+?)(-->\n|-->)/gs;
+	const REGEX_SVG_VIEWBOX = /(\sv|v)iewBox="(.+?)"/g;
+	const REGEX_SVG_TAG = /(<\/svg|<svg(.+?))>(\n|)/g;
```

Comment:
Pro-Tip™: all of these capturing groups (eg. `(...)`) look unreferenced when you use the regexps, so it is a good practice to make them non-capturing with `?:`: eg. `(?:flags-|\.svg$)` instead of `(flags-|\.svg$)`.